src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -309,20 +309,29 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
             await Task.FromResult(true);
         }
 
-        private static async Task HandleSignatureHelpRequest(
+        private static async Task HandleDocumentHighlightRequest(
             TextDocumentPosition textDocumentPosition,
-            RequestContext<SignatureHelp> requestContext)
+            RequestContext<DocumentHighlight[]> requestContext)
         {
             await Task.FromResult(true);
         }
+#endif
 
-        private static async Task HandleDocumentHighlightRequest(
+        private static async Task HandleSignatureHelpRequest(
```

Comment:
>HandleSignatureHelpRequest [](start = 34, length = 26)

I think you missed adding a check to only process if suggestions are enabled:
if (!WorkspaceService<SqlToolsSettings>.Instance.CurrentSettings.IsSuggestionsEnabled)
            {
                await Task.FromResult(true);
            }